cni/network/network.go

```diff
@@ -446,123 +448,157 @@ func (plugin *NetPlugin) Add(args *cniSkel.CmdArgs) error {
 			return fmt.Errorf("%w", err)
 		}
 
-		ipamAddResult.ncResponse, ipamAddResult.hostSubnetPrefix, er = plugin.multitenancyClient.GetContainerNetworkConfiguration(
+		ncResponses, hostSubnetPrefixes, er := plugin.multitenancyClient.GetNetworkContainersWithOrchestratorContext(
 			context.TODO(), nwCfg, k8sPodName, k8sNamespace)
-		if er != nil {
-			er = errors.Wrapf(er, "GetContainerNetworkConfiguration failed for podname %v namespace %v", k8sPodName, k8sNamespace)
-			log.Printf("%+v", er)
-			return er
-		}
 
-		ipamAddResult.ipv4Result = convertToCniResult(ipamAddResult.ncResponse, args.IfName)
+		log.Printf("hostSubnetPrefixes are %+v", hostSubnetPrefixes)
 
-		log.Printf("PrimaryInterfaceIdentifier: %v", ipamAddResult.hostSubnetPrefix.IP.String())
-	}
+		if ncResponses == nil {
+			log.Printf("CNS is old version, invoke old CNI API")
+			ipamAddResult.ncResponse, ipamAddResult.hostSubnetPrefix, er = plugin.multitenancyClient.GetNetworkContainerWithOrchestratorContext(
+				context.TODO(), nwCfg, k8sPodName, k8sNamespace)
+			if er != nil {
+				er = errors.Wrapf(er, "GetContainerNetworkConfiguration failed for podname %v namespace %v", k8sPodName, k8sNamespace)
+				log.Printf("%+v", er)
+				return er
+			}
+			ipamAddResult.ipv4Result = convertToCniResult(ipamAddResult.ncResponse, args.IfName)
+			ipamAddResults = append(ipamAddResults, ipamAddResult)
 
-	// Initialize values from network config.
-	networkID, err := plugin.getNetworkName(args.Netns, &ipamAddResult, nwCfg)
-	if err != nil {
-		log.Printf("[cni-net] Failed to extract network name from network config. error: %v", err)
-		return err
+			log.Printf("PrimaryInterfaceIdentifier: %v", ipamAddResult.hostSubnetPrefix.IP.String())
+		} else {
+			if er != nil {
+				er = errors.Wrapf(er, "GetNetworkContainersWithOrchestratorContext failed for podname %v namespace %v", k8sPodName, k8sNamespace)
+				log.Printf("%+v", er)
+				return er
+			}
+
+			var ncResponsesCopy [2]cns.GetNetworkContainerResponse
+			for i := 0; i < len(ncResponses); i++ {
```

Comment:
should use slice and use for range

Reply:
using range will cause github issue and for all for loop, we need to use this type...

Reply:
> using range will cause github issue

I am concerned that you don't understand why you have made the change that you made because this statement does not make sense